### PR TITLE
Improvement of node interface

### PR DESCRIPTION
**Pull request by pirower**
Hi, please let me know if you want me to incorporate all added features from the child Node_easy directly to Node. As I understood "Node" in the folder "easy" is already a facilitation to use message etc. from the base folder. Nether the less, Node doesn't implement any Ant+ devices (e.g heart rate montior) and so on. I added some features to enable unexperienced users to execute some functions (scan, connect to a heart rate monitor, disconnect, etc...) without much effort. In addition I put node.start into a thread to be able to execute commands (connect, disconnect, scan) while the ant+ dongle is running.
If you think it is to confusing and should be added directly into Node, please let me know

In node.py I added the function "get_capabilities" which returns the capabilities of the attached Ant+ dongle. This function is executed during the initialisation of Node. I changed the list of self.channels to a vector of fixed size ( self.channels = [None]*cap["max_channels"] ). Every time a new channel is created, the code looks up if there is still a free channel left. I also added a function to remove a channel (including unassignment and closing of the channel)

### CODE REVIEW

**Comment by pirower** [comment]
Hi Tigge, could you please provide me with some feedback. Thx

File: examples/scan.py [hunk 1]
```diff
@@ -50,28 +52,86 @@ def on_data(data):
...
+    channel_hrm2.set_rf_freq(57)
     try:
-        channel.open()
+        channel_scan.open()
+        time.sleep(10)
+        channel_scan.close()
+        channel_scan._unassign()
+        channel_hrm.open()
+        channel_hrm2.open()
+        #channel_scan.open()
```
Comment (Tigge) [inline examples/scan.py:128]: Remove line

**Comment by pirower** [comment]
Hi,
so you would see "plus" as another folder under ant/ (next to easy, base, fs and test)? Ant provides a document (device profile) for the different devices running under Ant+ to guarantee a standard message protocol (e.g. heart rate monitor, frequency sensor for a bike, watt sensor, etc etc..). All these devices have different data pages. Therefor you would need to set the device type (e.g. 120 for heart rate montior) in the channel and decode the data pages according to the device type. I could start with heart rate montior to set a base and keep it open to be completed by other device profiles.

**Comment by pirower** [comment]
Please take a look also in the commit for node_easy and tell me if you want the following functions/fucntionalities included in node:

def remove_deviceNum(self, deviceNum) #only makes sense if a deviceNum has been set in channel
def start(self) (as a thread)
def stop(self): (stop of the thread)
def scan(self, deviceType, timeout = 5, callback=None)

**Comment by Tigge** [comment]
> so you would see "plus" as another folder under ant/ (next to easy, base, fs and test)? 

Yes

> Ant provides a document (device profile) for the different devices running under Ant+ to guarantee a standard message protocol (e.g. heart rate monitor, frequency sensor for a bike, watt sensor, etc etc..). All these devices have different data pages. Therefor you would need to set the device type (e.g. 120 for heart rate montior) in the channel and decode the data pages according to the device type. I could start with heart rate montior to set a base and keep it open to be completed by other device profiles.

Yes, that is how it would be done. Sounds like a plan!

**Comment by Tigge** [comment]
> Please take a look also in the commit for node_easy and tell me if you want the following functions/fucntionalities included in node:
> 
> def remove_deviceNum(self, deviceNum) #only makes sense if a deviceNum has been set in channel
> def start(self) (as a thread)
> def stop(self): (stop of the thread)
> def scan(self, deviceType, timeout = 5, callback=None)

These might be better kept as an example (as it is in this PR now) on how to implement something like that then adding that functionality in node.

**Comment by Tigge** [comment]
The "Add capabilities to ant and make channels an array instead of a list", "Add new features in channel" and "Add features to node" are ready to commit I think (except the changes requested here). If you want to split this PR those could be merged quickly, but feel free to continue here If you like for the other stuff as well.

**Comment by tuna-f1sh** [comment]
Closing/unassigned channels and ANT+ device added with #76

**Pull request closed without merging**